pulsar-client/src/main/java/org/apache/pulsar/client/impl/BatchMessageContainerImpl.java

```diff
@@ -220,7 +220,7 @@ public OpSendMsg createOpSendMsg() throws IOException {
             messageMetadata.copyFrom(messages.get(0).getMessageBuilder());
             ByteBuf encryptedPayload = producer.encryptMessage(messageMetadata, getCompressedBatchMetadataAndPayload());
             ByteBufPair cmd = producer.sendMessage(producer.producerId, messageMetadata.getSequenceId(),
-                1, messageMetadata, encryptedPayload);
+                1, messages.get(0).getMessageId(), messageMetadata, encryptedPayload);
```

Comment:
This doesn't look right. Are you sure that this is the correct way to fix this?

Reply:
See #17195 . PIP-180 is for the shadow replicator. `null` should be passed here. I created another PR #17300 since I didn't notice this PR and I digged the reason why things broke.

Reply:
@Jason918 PTAL. isn't it `null` that should be passed here?

Reply:
@lhotari Actually, both is fine. ShadowReplicator won't use batch producing.